commands/java.py

```diff
@@ -48,6 +49,19 @@ def get_java_home():
         except CommandFailed:
             traceback.print_exc()
 
+    if is_linux():
+        try:
+            result = subprocess.run(
+                ["bash", "-c", "readlink -f $(which javac) | sed 's:/bin/javac::'"],
```

Comment:
This can be done in python without spawning a shell with `pathlib.Path(shutil.which("javac")).resolve().parents[1].as_posix()`. I would prefer we avoid spawning a shell unless we are doing something python can't easily do.

Reply:
That is a Pythonic way to do it. I will adjust the function based on your suggestion. Thanks for your review.